etcdserver/kv.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package etcdserver
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/coreos/etcd/pkg/fileutil"
+	dstorage "github.com/coreos/etcd/storage"
+)
+
+const databaseFilename = "db"
+
+type kv struct {
+	// dir to host KV file and incoming snapshot file
+	dir string
+	ig  dstorage.ConsistentIndexGetter
+
+	dstorage.ConsistentWatchableKV
+}
+
+func newKV(dir string, ig dstorage.ConsistentIndexGetter) *kv {
+	kv := &kv{
+		dir: dir,
+		ig:  ig,
+		ConsistentWatchableKV: dstorage.New(path.Join(dir, databaseFilename), ig),
+	}
+	if err := kv.ConsistentWatchableKV.Restore(); err != nil {
+		plog.Fatalf("v3 storage restore error: %v", err)
```

Comment:
can the fatal move back into server.go? I really don't like random exit points scattered around the codebase